### PR TITLE
SAAS-20313/afd-detection-mode

### PR DESCRIPTION
**Pull request by alphadev4**
https://github.com/aquasecurity/cloudsploit/pull/1737/files Merge this first

### CODE REVIEW

File: plugins/azure/frontdoor/frontDoorWafDetectionMode.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door WAF Detection Mode',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    description: 'Ensure that WAF policy for Azure Front Door is set to Detection mode.',
+    more_info: 'Azure Web Application Firewall (WAF) on Azure Front Door provides centralized protection of your web applications from common exploits and vulnerabilities. Web applications are increasingly targeted by malicious attacks that exploit commonly known vulnerabilities. It monitors and logs the request and its matched WAF rule to WAF logs.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorWafDetectionMode.js:9]:
```suggestion
    more_info: 'Web Application Firewall (WAF) on Front Door provides centralized protection of your web applications from common exploits and vulnerabilities. Web applications are increasingly targeted by malicious attacks that exploit commonly known vulnerabilities. It monitors and logs the request and its matched WAF rule to WAF logs.',
```

**Review by mehakseedat63 — APPROVED** [review]

**Comment by mehakseedat63** [comment]
Fix failing test

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]
Fix failing test

**Review by mehakseedat63 — APPROVED** [review]

File: exports.js [hunk 2]
```diff
@@ -992,11 +992,14 @@ module.exports = {
...
         'accessLogsEnabled'             : require(__dirname + '/plugins/azure/frontdoor/accessLogsEnabled.js'),
         'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js'),
+        'frontDoorWafDetectionMode'     : require(__dirname + '/plugins/azure/frontDoor/frontDoorWafDetectionMode.js'),
+        'frontDoorWafDetectionMode'     : require(__dirname + '/plugins/azure/frontDoor/frontDoorWafDetectionMode.js'),
         'frontDoorRequestBodyInspection': require(__dirname + '/plugins/azure/frontdoor/frontDoorRequestBodyInspection.js'),
         'frontDoorWafEnabled'           : require(__dirname + '/plugins/azure/frontdoor/frontDoorWafEnabled.js'),
         'frontDoorHttpsOnly'            : require(__dirname + '/plugins/azure/frontdoor/frontDoorHttpsOnly.js'),
         'botProtectionEnabled'          : require(__dirname + '/plugins/azure/frontdoor/botProtectionEnabled.js'),
 
+        'namespaceEncryptionAtRest'    : require(__dirname + '/plugins/azure/servicebus/namespaceEncryptionAtRest.js'),
```
Comment (alphadev4) [inline exports.js:1002]:
```suggestion
        'namespaceEncryptionAtRest'     : require(__dirname + '/plugins/azure/servicebus/namespaceEncryptionAtRest.js'),
```

**Pull request merged**